src/components/generic/server.hpp

```diff
@@ -19,7 +19,5 @@ class GenericServer
 
   GenericServer(){};
   GenericServer(std::shared_ptr<SubtypeService> sub_svc) : sub_svc(sub_svc){};
-
-private:
   std::shared_ptr<SubtypeService> sub_svc;
```

Comment:
this should probably stay private. If we need to access it from outside the class then we can always add an accessor function.